### ISSUE TITLE
Code Inspection - False Positive & Bad Suggestion for Private UserForm Variables

### ISSUE THREAD

**Issue by Stevefb** [rubberduck-vba/Rubberduck#1900]
Private variables declared at module level in userforms and which are only used in 1 sub / function are flagged by RD as an opportunity to reduce scope by moving the variable to the module that uses it.  

This suggestion does not seem to allow for situations where the variable is to an external sub to pass a parameter / variable to the userform.

**Comment by retailcoder** [comment]
A `Private` variable could not be accessed or written to by an external `Sub` procedure, can you clarify?

**Comment by Stevefb** [comment]
@retailcoder I'm not sure I understand you question.  This is for a userform to which I want to assign a specific document to a Private variable.  

To do this the Private variable is declared for the scope of the userform.  In the calling function:
```vb
Load myUserForm
myUserForm.setDocForThisUF(doc As Document)
myUserForm.Show
```
and in the userform:
```vb
Private docForUF as Document
Friend Function setDocForThisUF(doc As Document) As Document
Set docForUF = doc
End Function
```
Ouch, and now that I type this out something's wrong... docForUF is used at least twice, once by setDocForThisUF and the other time by whatever method in the UserForm is supposed to use it... drats, I'll have to go back and check...

**Comment by retailcoder** [comment]
I think I got it. Basically you have a private field and a public function to set it from outside the form, correct? If that field isn't used in another member of the form, the suggestion to reduce the variable's scope should be accompanied by a warning about that variable not being used (despite being assigned once).

If that field was encapsulated with a property, it would be a write-only property, which would fire up another warning.

In other words I'm not sure Rubberduck is wrong here. Can you show more of the form's code?

**Comment by Stevefb** [comment]
@retailcoder  Yes, that's where I was coming from. The suggestions offered by RD seem to have changed for this scenario and I'm no longer able to elicit the same suggestions.  

RD is now telling me (appropriately) that the function does not return a value, I don't recall seeing that inspection result previously --> heading in the right direction!  Thanks :)
